pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,291 @@
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/common/label"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+var _ = ginkgo.Describe("[Suite: e2e] NodePool STS Permissions", ginkgo.Ordered, label.HyperShift, label.E2E, func() {
+	var h *helper.H
+	var client *resources.Resources
+	var clusterNamespace string
+	var testNodePoolName string
+	var initialNodeCount int
+
+	nodePoolGVR := schema.GroupVersionResource{
+		Group:    "hypershift.openshift.io",
+		Version:  "v1beta1",
+		Resource: "nodepools",
+	}
+
+	ginkgo.BeforeAll(func() {
+		if !viper.GetBool(config.Hypershift) {
+			ginkgo.Skip("NodePool tests are only supported on HyperShift clusters")
+		}
+
+		h = helper.New()
+		client = h.AsUser("")
+
+		var nodeList corev1.NodeList
+		expect.NoError(client.List(context.Background(), &nodeList))
+
+		if len(nodeList.Items) == 0 {
+			ginkgo.Skip("No nodes found - cannot run NodePool tests")
+		}
+
+		initialNodeCount = len(nodeList.Items)
+
+		for _, node := range nodeList.Items {
+			if label, exists := node.Labels["hypershift.openshift.io/nodePool"]; exists {
+				parts := strings.Split(label, "-workers-")
+				if len(parts) >= 1 {
+					clusterNamespace = parts[0]
+					break
+				}
+			}
+		}
+
+		if clusterNamespace == "" {
+			ginkgo.Skip("Could not determine cluster namespace from node labels")
+		}
+
+		testNodePoolName = fmt.Sprintf("test-%d", time.Now().Unix()%100000)
+	})
+
+	ginkgo.AfterAll(func(ctx context.Context) {
+		if testNodePoolName != "" {
+			ginkgo.By("Cleaning up test NodePool")
+			err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+				Delete(ctx, testNodePoolName, metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				ginkgo.GinkgoLogr.Error(err, "Failed to cleanup test NodePool", "name", testNodePoolName)
+			}
+
+			ginkgo.By("Cleaning up test pods")
+			podList := &corev1.PodList{}
+			err = client.WithNamespace(h.CurrentProject()).List(ctx, podList)
+			if err == nil {
+				for _, pod := range podList.Items {
+					if strings.HasPrefix(pod.Name, "nodepool-test-") {
+						client.Delete(ctx, &pod)
+					}
+				}
+			}
+		}
+	})
+
+	ginkgo.It("should successfully create NodePool", func(ctx context.Context) {
+		ginkgo.By("Getting existing NodePool configuration")
+
+		existingNodePools, err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).List(ctx, metav1.ListOptions{})
+		expect.NoError(err, "Failed to list existing NodePools")
+		Expect(len(existingNodePools.Items)).To(BeNumerically(">", 0), "No existing NodePools found to reference")
```

Comment:
What would be the use case where there isn't a node pool?  And, if there isn't, can't we still proceed here?  (This may be a lack of my understanding, by the way.  Feel free to correct me :) )